kafka-ui-api/src/main/java/com/provectus/kafka/ui/client/BufSchemaRegistryClient.java

```diff
@@ -1,7 +1,140 @@
 package com.provectus.kafka.ui.client;
 
+import com.google.protobuf.DescriptorProtos.FileDescriptorProto;
+import com.google.protobuf.DescriptorProtos.FileDescriptorSet;
+import com.google.protobuf.Descriptors.Descriptor;
+import com.google.protobuf.Descriptors.DescriptorValidationException;
+import com.google.protobuf.Descriptors.FileDescriptor;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.provectus.kafka.ui.proto.gen.buf.alpha.image.v1.Image;
+import com.provectus.kafka.ui.proto.gen.buf.alpha.registry.v1alpha1.GetImageRequest;
+import com.provectus.kafka.ui.proto.gen.buf.alpha.registry.v1alpha1.GetImageResponse;
+import com.provectus.kafka.ui.proto.gen.buf.alpha.registry.v1alpha1.ImageServiceGrpc;
+import io.grpc.Channel;
+import io.grpc.ManagedChannelBuilder;
+import io.grpc.Metadata;
+import io.grpc.StatusRuntimeException;
+import io.grpc.stub.MetadataUtils;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
 public class BufSchemaRegistryClient {
-  public BufSchemaRegistryClient() {}
+  private ImageServiceGrpc.ImageServiceBlockingStub bufClient;
+
+  public BufSchemaRegistryClient(String host, int port, String apiKey) {
+    this(ManagedChannelBuilder.forAddress(String.format("api.%s", host), port).useTransportSecurity(), apiKey);
+  }
+
+  public BufSchemaRegistryClient(ManagedChannelBuilder<?> channelBuilder, String apiKey) {
+    Channel channel = channelBuilder.build();
+    bufClient = ImageServiceGrpc.newBlockingStub(channel);
+
+    Metadata headers = new Metadata();
+    Metadata.Key<String> bearerKey = Metadata.Key.of("authorization", Metadata.ASCII_STRING_MARSHALLER);
+    headers.put(bearerKey, String.format("Bearer %s", apiKey));
+
+    bufClient = bufClient.withInterceptors(MetadataUtils.newAttachHeadersInterceptor(headers));
+  }
+
+  @Nullable
```

Comment:
since you aren't implementing an interface here... you could do the better thing and return `Optional<Descriptor>` instead. Nullable is a bad crutch